src/main/java/com/rackspacecloud/metrics/ingestionservice/influxdb/InfluxDBHelper.java

```diff
@@ -341,13 +328,12 @@ public void ingestToInfluxDb(
             // Enable or disable using
             backupService.writeToBackup(payload, new URL(baseUrl), databaseName, retPolicyName);
             influxDB.write(databaseName, retPolicyName, InfluxDB.ConsistencyLevel.ONE, TimeUnit.SECONDS, payload);
-        }
-        catch(InfluxDBException.PointsBeyondRetentionPolicyException ex) {
+        } catch(IOException ex) {
             log.error("Write failed for the payload. baseURL: [{}], databaseName: [{}], ret-policy: [{}]",
                     baseUrl, databaseName, retPolicyName);
             throw ex;
         }
```

Comment:
...but also keep this `IOException` as a broader catch. Instead of re-throwing the exception as-is, throw a new `IngestFailedException`. In fact, `InvalidDataException` could extend `IngestFailedException` since it indicates a specialized form of ingest failure.